### ISSUE TITLE
Swaylock exits/crashes after about two hours

### ISSUE THREAD

**Issue by SuperTux88** [swaywm/swaylock#126]
I tried wayland/sway in the last few days, and after locking my PC and leaving, I noticed, that it was unlocked after I returned. I wrote a little wrapper script to collect some information:

* it crashes more or less exactly after two hours and a few seconds (seconds vary, today it was 2 hour and 51 seconds, yesterday it crashed after 2 hours and 53 seconds)
* it exits with an exit code `0` ... so I'm not even sure if it's a crash? (I wanted to write a loop that restarts it, if it doesn't exit cleanly, but with the exit code it's not possible to see if it "crashed" or was unlocked with a password)
* the only error it prints is `wl_display@1: error 0: invalid object 40` or `wl_display@1: error 0: invalid object 41`, but I have no idea what this means? Starting it with `--debug` didn't print anything useful (all lines with a timestamp are from the start, not from when it exited):
    ```
    2020-01-14 02:55:57 - [swaylock-1.4/main.c:1101] Parsing CLI Args
    2020-01-14 02:55:57 - [swaylock-1.4/main.c:278] output name is DP-3
    2020-01-14 02:55:57 - [swaylock-1.4/main.c:278] output name is DP-2
    2020-01-14 02:55:57 - [swaylock-1.4/main.c:278] output name is DP-1
    wl_display@1: error 0: invalid object 40
    ```
* it only crashes once, if I restart it after it exited the first time, it keeps running for many hours, until I unlock it (so my current workaround instead of the planned loop is now just a little script that has the same swaylock line twice, the first exits after two hours, and the second one works then as expected, only disadvantage: If I want to unlock it before two hours, I need to enter the password twice).
* I thought that maybe swayidle causes some problems (although I didn't have anything set for two hours, but it was weird that it always exits just after two hours of idle, and only once), but I don't have any swayidle commands running anymore, and it still exits after two hours.

Is there anything more I can do to see why it exits or to get more debug output? A lockscreen that just exits cleanly without entering a password is not really useful ...

**Comment by emersion** [comment]
Duplicate of https://github.com/swaywm/sway/issues/3625

**Comment by emersion** [comment]
Oh, wait, the cause seems a little bit different. Can you run swaylock with `WAYLAND_DEBUG=1`, reproduce the bug and post the output?

**Comment by SuperTux88** [comment]
I saw the linked issue before, but I don't change any monitors while it's locked, and also the error message looked different. But I saw that that is already fixed in latest wayland code, so I updated wayland to git-code and I'll see if it fixes it. I'll also run it with `WAYLAND_DEBUG=1` next time I lock my screen, so I can provide more debug output if it still fails.

**Comment by SuperTux88** [comment]
OK, even with the latest wayland from git, it still exits after two hours. Here is the output with `WAYLAND_DEBUG=1`:

```
2020-01-15 04:28:17 - [swaylock-1.4/main.c:1101] Parsing CLI Args
[1286102.003]  -> wl_display@1.get_registry(new id wl_registry@2)
[1286102.013]  -> wl_display@1.sync(new id wl_callback@3)
[1286102.064] wl_display@1.delete_id(3)
[1286102.083] wl_registry@2.global(1, "wl_shm", 1)
[1286102.092]  -> wl_registry@2.bind(1, "wl_shm", 1, new id [unknown]@4)
[1286102.100] wl_registry@2.global(2, "zwp_linux_dmabuf_v1", 3)
[1286102.105] wl_registry@2.global(3, "wl_drm", 2)
[1286102.109] wl_registry@2.global(4, "wl_compositor", 4)
[1286102.113]  -> wl_registry@2.bind(4, "wl_compositor", 3, new id [unknown]@5)
[1286102.119] wl_registry@2.global(5, "wl_subcompositor", 1)
[1286102.123]  -> wl_registry@2.bind(5, "wl_subcompositor", 1, new id [unknown]@6)
[1286102.128] wl_registry@2.global(6, "wl_data_device_manager", 3)
[1286102.132] wl_registry@2.global(7, "zwlr_gamma_control_manager_v1", 1)
[1286102.136] wl_registry@2.global(8, "gtk_primary_selection_device_manager", 1)
[1286102.140] wl_registry@2.global(9, "zxdg_output_manager_v1", 3)
[1286102.144]  -> wl_registry@2.bind(9, "zxdg_output_manager_v1", 2, new id [unknown]@7)
[1286102.149] wl_registry@2.global(10, "org_kde_kwin_idle", 1)
[1286102.153] wl_registry@2.global(11, "zwp_idle_inhibit_manager_v1", 1)
[1286102.158] wl_registry@2.global(12, "zwlr_layer_shell_v1", 2)
[1286102.162]  -> wl_registry@2.bind(12, "zwlr_layer_shell_v1", 1, new id [unknown]@8)
[1286102.168] wl_registry@2.global(13, "xdg_wm_base", 2)
[1286102.172] wl_registry@2.global(14, "zwp_tablet_manager_v2", 1)
[1286102.176] wl_registry@2.global(15, "org_kde_kwin_server_decoration_manager", 1)
[1286102.180] wl_registry@2.global(16, "zxdg_decoration_manager_v1", 1)
[1286102.185] wl_registry@2.global(17, "zwp_relative_pointer_manager_v1", 1)
[1286102.189] wl_registry@2.global(18, "zwp_pointer_constraints_v1", 1)
[1286102.193] wl_registry@2.global(19, "wp_presentation", 1)
[1286102.197] wl_registry@2.global(20, "zwlr_output_manager_v1", 1)
[1286102.201] wl_registry@2.global(21, "zwlr_export_dmabuf_manager_v1", 1)
[1286102.205] wl_registry@2.global(22, "zwlr_screencopy_manager_v1", 2)
[1286102.209] wl_registry@2.global(23, "zwlr_data_control_manager_v1", 2)
[1286102.213] wl_registry@2.global(24, "zwp_primary_selection_device_manager_v1", 1)
[1286102.217] wl_registry@2.global(25, "zwp_virtual_keyboard_manager_v1", 1)
[1286102.222] wl_registry@2.global(26, "zwlr_virtual_pointer_manager_v1", 1)
[1286102.226] wl_registry@2.global(27, "zwlr_input_inhibit_manager_v1", 1)
[1286102.230]  -> wl_registry@2.bind(27, "zwlr_input_inhibit_manager_v1", 1, new id [unknown]@9)
[1286102.236] wl_registry@2.global(28, "wl_seat", 7)
[1286102.240]  -> wl_registry@2.bind(28, "wl_seat", 3, new id [unknown]@10)
[1286102.247] wl_registry@2.global(29, "zwp_pointer_gestures_v1", 1)
[1286102.252] wl_registry@2.global(30, "wl_output", 3)
[1286102.256]  -> wl_registry@2.bind(30, "wl_output", 3, new id [unknown]@11)
[1286102.262] wl_registry@2.global(31, "wl_output", 3)
[1286102.267]  -> wl_registry@2.bind(31, "wl_output", 3, new id [unknown]@12)
[1286102.273] wl_registry@2.global(32, "wl_output", 3)
[1286102.277]  -> wl_registry@2.bind(32, "wl_output", 3, new id [unknown]@13)
[1286102.283] wl_callback@3.done(26688)
[1286102.286]  -> zwlr_input_inhibit_manager_v1@9.get_inhibitor(new id zwlr_input_inhibitor_v1@3)
[1286102.290]  -> wl_display@1.sync(new id wl_callback@14)
[1286102.332] wl_display@1.delete_id(14)
[1286102.335] wl_seat@10.name("seat0")
[1286102.337] wl_seat@10.capabilities(3)
[1286102.338]  -> wl_seat@10.get_pointer(new id wl_pointer@15)
[1286102.342]  -> wl_seat@10.get_keyboard(new id wl_keyboard@16)
[1286102.345] wl_output@11.geometry(0, 0, 640, 400, 0, "Dell Inc.", "DELL U3014", 0)
[1286102.353] wl_output@11.mode(1, 2560, 1600, 59972)
[1286102.357] wl_output@11.scale(1)
[1286102.360] wl_output@11.done()
[1286102.362] wl_output@12.geometry(0, 0, 640, 400, 0, "Dell Inc.", "DELL U3014", 0)
[1286102.369] wl_output@12.mode(1, 2560, 1600, 59972)
[1286102.373] wl_output@12.scale(1)
[1286102.376] wl_output@12.done()
[1286102.377] wl_output@13.geometry(0, 0, 640, 400, 0, "Hewlett Packard", "HP ZR30w", 0)
[1286102.385] wl_output@13.mode(1, 2560, 1600, 59972)
[1286102.388] wl_output@13.scale(1)
[1286102.390] wl_output@13.done()
[1286102.391] wl_callback@14.done(26690)
[1286102.393]  -> zxdg_output_manager_v1@7.get_xdg_output(new id zxdg_output_v1@14, wl_output@13)
[1286102.396]  -> zxdg_output_manager_v1@7.get_xdg_output(new id zxdg_output_v1@17, wl_output@12)
[1286102.399]  -> zxdg_output_manager_v1@7.get_xdg_output(new id zxdg_output_v1@18, wl_output@11)
[1286102.402]  -> wl_display@1.sync(new id wl_callback@19)
[1286102.469] wl_display@1.delete_id(19)
[1286102.471] wl_keyboard@16.keymap(1, fd 7, 48877)
[1286103.667] zxdg_output_v1@14.name("DP-3")
2020-01-15 04:28:17 - [swaylock-1.4/main.c:278] output name is DP-3
[1286103.677] zxdg_output_v1@14.description("Hewlett Packard HP ZR30w CN40520374 (DP-3)")
[1286103.678] zxdg_output_v1@14.logical_position(5120, 0)
[1286103.681] zxdg_output_v1@14.logical_size(2560, 1600)
[1286103.683] zxdg_output_v1@14.done()
[1286103.684] wl_output@13.done()
[1286103.685] zxdg_output_v1@17.name("DP-2")
2020-01-15 04:28:17 - [swaylock-1.4/main.c:278] output name is DP-2
[1286103.689] zxdg_output_v1@17.description("Dell Inc. DELL U3014 P1V6N3B8329L (DP-2)")
[1286103.690] zxdg_output_v1@17.logical_position(2560, 0)
[1286103.693] zxdg_output_v1@17.logical_size(2560, 1600)
[1286103.696] zxdg_output_v1@17.done()
[1286103.697] wl_output@12.done()
[1286103.699] zxdg_output_v1@18.name("DP-1")
2020-01-15 04:28:17 - [swaylock-1.4/main.c:278] output name is DP-1
[1286103.704] zxdg_output_v1@18.description("Dell Inc. DELL U3014 P1V6N3BG276L (DP-1)")
[1286103.706] zxdg_output_v1@18.logical_position(0, 0)
[1286103.710] zxdg_output_v1@18.logical_size(2560, 1600)
[1286103.714] zxdg_output_v1@18.done()
[1286103.715] wl_output@11.done()
[1286103.717] wl_callback@19.done(26690)
[1286103.722]  -> wl_compositor@5.create_surface(new id wl_surface@19)
[1286103.725]  -> wl_compositor@5.create_surface(new id wl_surface@20)
[1286103.728]  -> wl_subcompositor@6.get_subsurface(new id wl_subsurface@21, wl_surface@20, wl_surface@19)
[1286103.734]  -> wl_subsurface@21.set_sync()
[1286103.736]  -> zwlr_layer_shell_v1@8.get_layer_surface(new id zwlr_layer_surface_v1@22, wl_surface@19, wl_output@13, 3, "lockscreen")
[1286103.743]  -> zwlr_layer_surface_v1@22.set_size(0, 0)
[1286103.747]  -> zwlr_layer_surface_v1@22.set_anchor(15)
[1286103.749]  -> zwlr_layer_surface_v1@22.set_exclusive_zone(-1)
[1286103.751]  -> zwlr_layer_surface_v1@22.set_keyboard_interactivity(1)
[1286103.753]  -> wl_compositor@5.create_region(new id wl_region@23)
[1286103.756]  -> wl_region@23.add(0, 0, 2147483647, 2147483647)
[1286103.760]  -> wl_surface@19.set_opaque_region(wl_region@23)
[1286103.762]  -> wl_region@23.destroy()
[1286103.764]  -> wl_surface@19.commit()
[1286103.766]  -> wl_compositor@5.create_surface(new id wl_surface@24)
[1286103.769]  -> wl_compositor@5.create_surface(new id wl_surface@25)
[1286103.771]  -> wl_subcompositor@6.get_subsurface(new id wl_subsurface@26, wl_surface@25, wl_surface@24)
[1286103.775]  -> wl_subsurface@26.set_sync()
[1286103.776]  -> zwlr_layer_shell_v1@8.get_layer_surface(new id zwlr_layer_surface_v1@27, wl_surface@24, wl_output@12, 3, "lockscreen")
[1286103.781]  -> zwlr_layer_surface_v1@27.set_size(0, 0)
[1286103.783]  -> zwlr_layer_surface_v1@27.set_anchor(15)
[1286103.785]  -> zwlr_layer_surface_v1@27.set_exclusive_zone(-1)
[1286103.787]  -> zwlr_layer_surface_v1@27.set_keyboard_interactivity(1)
[1286103.789]  -> wl_compositor@5.create_region(new id wl_region@28)
[1286103.792]  -> wl_region@28.add(0, 0, 2147483647, 2147483647)
[1286103.795]  -> wl_surface@24.set_opaque_region(wl_region@28)
[1286103.797]  -> wl_region@28.destroy()
[1286103.798]  -> wl_surface@24.commit()
[1286103.800]  -> wl_compositor@5.create_surface(new id wl_surface@29)
[1286103.802]  -> wl_compositor@5.create_surface(new id wl_surface@30)
[1286103.803]  -> wl_subcompositor@6.get_subsurface(new id wl_subsurface@31, wl_surface@30, wl_surface@29)
[1286103.806]  -> wl_subsurface@31.set_sync()
[1286103.808]  -> zwlr_layer_shell_v1@8.get_layer_surface(new id zwlr_layer_surface_v1@32, wl_surface@29, wl_output@11, 3, "lockscreen")
[1286103.813]  -> zwlr_layer_surface_v1@32.set_size(0, 0)
[1286103.815]  -> zwlr_layer_surface_v1@32.set_anchor(15)
[1286103.816]  -> zwlr_layer_surface_v1@32.set_exclusive_zone(-1)
[1286103.818]  -> zwlr_layer_surface_v1@32.set_keyboard_interactivity(1)
[1286103.820]  -> wl_compositor@5.create_region(new id wl_region@33)
[1286103.822]  -> wl_region@33.add(0, 0, 2147483647, 2147483647)
[1286103.826]  -> wl_surface@29.set_opaque_region(wl_region@33)
[1286103.828]  -> wl_region@33.destroy()
[1286103.829]  -> wl_surface@29.commit()
[1286103.877] wl_display@1.delete_id(23)
[1286103.879] wl_display@1.delete_id(28)
[1286103.881] wl_display@1.delete_id(33)
[1286103.882] zwlr_layer_surface_v1@22.configure(26691, 2560, 1600)
[1286103.887]  -> zwlr_layer_surface_v1@22.ack_configure(26691)
[1286103.909]  -> wl_shm@4.create_pool(new id wl_shm_pool@33, fd 8, 16384000)
[1286103.914]  -> wl_shm_pool@33.create_buffer(new id wl_buffer@28, 0, 2560, 1600, 10240, 0)
[1286103.921]  -> wl_shm_pool@33.destroy()
[1286108.584]  -> wl_surface@19.set_buffer_scale(1)
[1286108.594]  -> wl_surface@19.attach(wl_buffer@28, 0, 0)
[1286108.598]  -> wl_surface@19.damage(0, 0, 2560, 1600)
[1286108.602]  -> wl_surface@19.commit()
[1286108.605]  -> wl_subsurface@21.set_position(1220, 740)
[1286108.635]  -> wl_shm@4.create_pool(new id wl_shm_pool@23, fd 9, 115200)
[1286108.639]  -> wl_shm_pool@23.create_buffer(new id wl_buffer@34, 0, 120, 240, 480, 0)
[1286108.646]  -> wl_shm_pool@23.destroy()
[1286108.655]  -> wl_surface@20.attach(nil, 0, 0)
[1286108.658]  -> wl_surface@20.commit()
[1286108.708]  -> wl_surface@20.set_buffer_scale(1)
[1286108.710]  -> wl_surface@20.attach(wl_buffer@34, 0, 0)
[1286108.714]  -> wl_surface@20.damage(0, 0, 120, 240)
[1286108.720]  -> wl_surface@20.commit()
[1286108.722]  -> wl_surface@19.commit()
[1286108.725] zwlr_layer_surface_v1@27.configure(26692, 2560, 1600)
[1286108.729]  -> zwlr_layer_surface_v1@27.ack_configure(26692)
[1286108.739]  -> wl_shm@4.create_pool(new id wl_shm_pool@35, fd 10, 16384000)
[1286108.743]  -> wl_shm_pool@35.create_buffer(new id wl_buffer@36, 0, 2560, 1600, 10240, 0)
[1286108.748]  -> wl_shm_pool@35.destroy()
[1286113.445]  -> wl_surface@24.set_buffer_scale(1)
[1286113.456]  -> wl_surface@24.attach(wl_buffer@36, 0, 0)
[1286113.460]  -> wl_surface@24.damage(0, 0, 2560, 1600)
[1286113.465]  -> wl_surface@24.commit()
[1286113.467]  -> wl_subsurface@26.set_position(1220, 740)
[1286113.498]  -> wl_shm@4.create_pool(new id wl_shm_pool@37, fd 11, 115200)
[1286113.502]  -> wl_shm_pool@37.create_buffer(new id wl_buffer@38, 0, 120, 240, 480, 0)
[1286113.509]  -> wl_shm_pool@37.destroy()
[1286113.519]  -> wl_surface@25.attach(nil, 0, 0)
[1286113.522]  -> wl_surface@25.commit()
[1286113.569]  -> wl_surface@25.set_buffer_scale(1)
[1286113.572]  -> wl_surface@25.attach(wl_buffer@38, 0, 0)
[1286113.574]  -> wl_surface@25.damage(0, 0, 120, 240)
[1286113.578]  -> wl_surface@25.commit()
[1286113.579]  -> wl_surface@24.commit()
[1286113.581] zwlr_layer_surface_v1@32.configure(26693, 2560, 1600)
[1286113.586]  -> zwlr_layer_surface_v1@32.ack_configure(26693)
[1286113.597]  -> wl_shm@4.create_pool(new id wl_shm_pool@39, fd 12, 16384000)
[1286113.601]  -> wl_shm_pool@39.create_buffer(new id wl_buffer@40, 0, 2560, 1600, 10240, 0)
[1286113.606]  -> wl_shm_pool@39.destroy()
[1286118.247]  -> wl_surface@29.set_buffer_scale(1)
[1286118.256]  -> wl_surface@29.attach(wl_buffer@40, 0, 0)
[1286118.260]  -> wl_surface@29.damage(0, 0, 2560, 1600)
[1286118.263]  -> wl_surface@29.commit()
[1286118.264]  -> wl_subsurface@31.set_position(1220, 740)
[1286118.291]  -> wl_shm@4.create_pool(new id wl_shm_pool@41, fd 13, 115200)
[1286118.295]  -> wl_shm_pool@41.create_buffer(new id wl_buffer@42, 0, 120, 240, 480, 0)
[1286118.299]  -> wl_shm_pool@41.destroy()
[1286118.305]  -> wl_surface@30.attach(nil, 0, 0)
[1286118.308]  -> wl_surface@30.commit()
[1286118.352]  -> wl_surface@30.set_buffer_scale(1)
[1286118.355]  -> wl_surface@30.attach(wl_buffer@42, 0, 0)
[1286118.357]  -> wl_surface@30.damage(0, 0, 120, 240)
[1286118.360]  -> wl_surface@30.commit()
[1286118.361]  -> wl_surface@29.commit()
[1286130.033] wl_display@1.delete_id(33)
[1286130.046] wl_display@1.delete_id(23)
[1286130.048] wl_display@1.delete_id(35)
[1286130.049] wl_display@1.delete_id(37)
[1286130.050] wl_display@1.delete_id(39)
[1286130.052] wl_display@1.delete_id(41)
[1286130.054] wl_buffer@28.release()
[1286130.056] wl_keyboard@16.enter(26694, wl_surface@19, array)
[1286130.060] wl_keyboard@16.modifiers(26695, 0, 0, 16, 0)
[1286130.075] wl_buffer@34.release()
[1286130.077] wl_buffer@36.release()
[1286130.078] wl_pointer@15.enter(26696, wl_surface@24, 939.953125, 853.859375)
[1286130.084]  -> wl_pointer@15.set_cursor(26696, nil, 0, 0)
[1286130.090] wl_keyboard@16.leave(26697, wl_surface@19)
[1286130.092] wl_keyboard@16.enter(26698, wl_surface@24, array)
[1286130.095] wl_keyboard@16.modifiers(26699, 0, 0, 16, 0)
[1286130.099] wl_buffer@38.release()
[1286130.100] wl_buffer@40.release()
[1286130.101] wl_keyboard@16.leave(26700, wl_surface@24)
[1286130.103] wl_keyboard@16.enter(26701, wl_surface@29, array)
[1286130.106] wl_keyboard@16.modifiers(26702, 0, 0, 16, 0)
[1286130.110] wl_buffer@42.release()
[1286179.594] wl_keyboard@16.key(26703, 21991726, 28, 0)
[1286724.408] wl_keyboard@16.leave(26704, wl_surface@29)
[1286724.420] wl_keyboard@16.enter(26705, wl_surface@19, array)
[1286724.423] wl_keyboard@16.modifiers(26706, 0, 0, 16, 0)
[1286724.429] wl_keyboard@16.leave(26707, wl_surface@19)
[1286724.432] wl_keyboard@16.enter(26708, wl_surface@29, array)
[1286724.434] wl_keyboard@16.modifiers(26709, 0, 0, 16, 0)
[1286724.438] wl_keyboard@16.leave(26710, wl_surface@29)
[1286724.440] wl_keyboard@16.enter(26711, wl_surface@24, array)
[1286724.442] wl_keyboard@16.modifiers(26712, 0, 0, 16, 0)
[1290296.156] wl_pointer@15.leave(26713, wl_surface@24)
[1290296.169] wl_keyboard@16.leave(26714, wl_surface@24)
[1290296.173] wl_keyboard@16.enter(26715, wl_surface@29, array)
[1290296.177] wl_keyboard@16.modifiers(26716, 0, 0, 16, 0)
[1290296.186] zwlr_layer_surface_v1@27.closed()
[1290296.192]  -> zwlr_layer_surface_v1@27.destroy()
[1290296.196]  -> wl_surface@24.destroy()
[1290296.198]  -> wl_buffer@36.destroy()
[1290296.788] wl_registry@2.global_remove(31)
[1290303.903] wl_display@1.delete_id(27)
[1290303.911] wl_display@1.delete_id(24)
[1290303.913] wl_display@1.delete_id(36)
[1290303.916] wl_pointer@15.enter(26718, wl_surface@29, 2559.000000, 853.859375)
[1290303.922]  -> wl_pointer@15.set_cursor(26718, nil, 0, 0)
[1294826.555] wl_pointer@15.leave(26719, wl_surface@29)
[1294826.595] wl_keyboard@16.leave(26720, wl_surface@29)
[1294826.599] wl_keyboard@16.enter(26721, wl_surface@19, array)
[1294826.607] wl_keyboard@16.modifiers(26722, 0, 0, 16, 0)
[1294826.629] zwlr_layer_surface_v1@32.closed()
[1294826.632]  -> zwlr_layer_surface_v1@32.destroy()
[1294826.635]  -> wl_surface@29.destroy()
[1294826.637]  -> wl_buffer@40.destroy()
[1294827.217] wl_registry@2.global_remove(30)
[1294833.970] wl_display@1.delete_id(32)
[1294833.985] wl_display@1.delete_id(29)
[1294833.987] wl_display@1.delete_id(40)
[1294833.988] wl_pointer@15.enter(26724, wl_surface@19, 0.000000, 853.859375)
[1294833.994]  -> wl_pointer@15.set_cursor(26724, nil, 0, 0)
[1294990.553] wl_keyboard@16.key(26725, 22000537, 55, 1)
[1294990.568]  -> wl_surface@19.frame(new id wl_callback@40)
[1294990.571]  -> wl_surface@19.commit()
[1294998.161] wl_display@1.delete_id(40)
[1294998.165] wl_callback@40.done(22000545)
[1294998.168]  -> wl_surface@19.frame(new id wl_callback@40)
[1294998.171]  -> wl_subsurface@21.set_position(1220, 740)
[1294998.174]  -> wl_surface@20.attach(nil, 0, 0)
[1294998.177]  -> wl_surface@20.commit()
[1294998.553]  -> wl_surface@20.set_buffer_scale(1)
[1294998.558]  -> wl_surface@20.attach(wl_buffer@34, 0, 0)
[1294998.562]  -> wl_surface@20.damage(0, 0, 120, 240)
[1294998.566]  -> wl_surface@20.commit()
[1294998.567]  -> wl_surface@19.commit()
[1294998.630] wl_buffer@34.release()
[1295014.737] wl_display@1.delete_id(40)
[1295014.743] wl_callback@40.done(22000561)
[1297993.734]  -> wl_surface@19.frame(new id wl_callback@40)
[1297993.749]  -> wl_surface@19.commit()
[1297999.615] wl_display@1.delete_id(40)
[1297999.622] wl_callback@40.done(22003546)
[1297999.624]  -> wl_surface@19.frame(new id wl_callback@40)
[1297999.627]  -> wl_subsurface@21.set_position(1220, 740)
[1297999.633]  -> wl_surface@20.attach(nil, 0, 0)
[1297999.636]  -> wl_surface@20.commit()
[1297999.654]  -> wl_surface@20.set_buffer_scale(1)
[1297999.657]  -> wl_surface@20.attach(wl_buffer@34, 0, 0)
[1297999.660]  -> wl_surface@20.damage(0, 0, 120, 240)
[1297999.663]  -> wl_surface@20.commit()
[1297999.664]  -> wl_surface@19.commit()
[1297999.714] wl_buffer@34.release()
[1298016.166] wl_display@1.delete_id(40)
[1298016.172] wl_callback@40.done(22003563)
[1304998.161]  -> wl_surface@19.frame(new id wl_callback@40)
[1304998.172]  -> wl_surface@19.commit()
[1305002.916] wl_display@1.delete_id(40)
[1305002.927] wl_callback@40.done(22010550)
[1305002.931]  -> wl_surface@19.frame(new id wl_callback@40)
[1305002.935]  -> wl_subsurface@21.set_position(1220, 740)
[1305002.938]  -> wl_surface@20.attach(nil, 0, 0)
[1305002.940]  -> wl_surface@20.commit()
[1305008.906]  -> wl_surface@20.set_buffer_scale(1)
[1305008.913]  -> wl_surface@20.attach(wl_buffer@34, 0, 0)
[1305008.916]  -> wl_surface@20.damage(0, 0, 120, 240)
[1305008.919]  -> wl_surface@20.commit()
[1305008.921]  -> wl_surface@19.commit()
[1305008.995] wl_buffer@34.release()
[1305019.500] wl_display@1.delete_id(40)
[1305019.508] wl_callback@40.done(22010566)
[1308000.902]  -> wl_surface@19.frame(new id wl_callback@40)
[1308000.915]  -> wl_surface@19.commit()
[1308004.283] wl_display@1.delete_id(40)
[1308004.287] wl_callback@40.done(22013551)
[1308004.290]  -> wl_surface@19.frame(new id wl_callback@40)
[1308004.293]  -> wl_subsurface@21.set_position(1220, 740)
[1308004.297]  -> wl_surface@20.attach(nil, 0, 0)
[1308004.300]  -> wl_surface@20.commit()
[1308004.319]  -> wl_surface@20.set_buffer_scale(1)
[1308004.322]  -> wl_surface@20.attach(wl_buffer@34, 0, 0)
[1308004.326]  -> wl_surface@20.damage(0, 0, 120, 240)
[1308004.330]  -> wl_surface@20.commit()
[1308004.331]  -> wl_surface@19.commit()
[1308004.392] wl_buffer@34.release()
[1308020.846] wl_display@1.delete_id(40)
[1308020.853] wl_callback@40.done(22013567)
[4242303.784] wl_seat@10.capabilities(3)
[4242303.796]  -> wl_pointer@15.release()
[4242303.801]  -> wl_keyboard@16.release()
[4242303.803]  -> wl_seat@10.get_pointer(new id wl_pointer@40)
[4242303.806]  -> wl_seat@10.get_keyboard(new id wl_keyboard@29)
[4242314.857] wl_display@1.delete_id(15)
[4242314.866] wl_display@1.delete_id(16)
[4242314.869] wl_seat@10.capabilities(2)
[4242314.873]  -> wl_pointer@40.release()
[4242314.876]  -> wl_keyboard@29.release()
[4242314.878]  -> wl_seat@10.get_keyboard(new id wl_keyboard@16)
[4242314.965] wl_display@1.error(wl_display@1, 0, "invalid object 40")
wl_display@1: error 0: invalid object 40
```

**Comment by emersion** [comment]
Definitely not the same bug.

**Comment by emersion** [comment]
I believe this is fixed by https://github.com/swaywm/wlroots/pull/2228

**Comment by markstos** [comment]
@emersion I recommend closing this since it's believed to be fixed.

**Comment by emersion** [comment]
Please open a new issue if you can still reproduce.
